src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/Locations/Queries/GetProviderLocationDetails/GetProviderLocationDetailsQueryHandler.cs

```diff
@@ -18,8 +19,14 @@ public GetProviderLocationDetailsQueryHandler(IRoatpCourseManagementApiClient<Ro
 
         public async Task<GetProviderLocationDetailsQueryResult> Handle(GetProviderLocationDetailsQuery request, CancellationToken cancellationToken)
         {
-            var response = await _courseManagementApiClient.Get<ProviderLocationModel>(request);
-            return new GetProviderLocationDetailsQueryResult() { ProviderLocation = response };
+            var response = await _courseManagementApiClient.GetWithResponseCode<ProviderLocationModel>(request);
+
+            if (response.StatusCode == HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+
```

Comment:
After checking for not found, we should still raise an exception for any other bad responses by using `EnsureSuccessStatus`